execution/src/posix/execution.cc

```diff
@@ -463,11 +448,26 @@ namespace leatherman { namespace execution {
                                             options[execution_options::inherit_locale]);
         auto envp = to_exec_arg(&variables);
 
+        // Set all open file descriptors above stderr to close on exec
+        auto max_desc_limit = get_max_descriptor_limit();
```

Comment:
@peterhuene does this seem like a sane replacement for closing the file descriptors in the child?